core/server/data/validation/index.js

```diff
@@ -42,6 +43,27 @@ validator.extend('isTimezone', function isTimezone(str) {
     return moment.tz.zone(str) ? true : false;
 });
 
+validator.extend('isEmptyOrWebsite', function isEmptyOrWebsite(str) {
```

Comment:
> The website input on the user profile page is supposed to be only a website, which clearly doesn't include query strings for example.

Are we sure query strings aren't valid for a website address? Eg. http://somesocialsite.com/?user=myusername is perfectly valid IMO. I think the thing we need to protect against is injected scripts which will mostly include `<` and `>` characters.